pkg/code-generator/codegen/templates/resource_template.go

```diff
@@ -38,12 +38,17 @@ var (
 {{- if $.IsCustom }}
 	_ resources.CustomInputResource = new({{ .Name }})
 {{- else }}
-	_ resources.InputResource = new({{ .Name }})
+	_ resources.HashableInputResource = new({{ .Name }})
 {{- end }}
 {{- else }}
 	_ resources.Resource = new({{ .Name }})
 {{- end }}
 )
+{{- if not $.IsCustom }}
```

Comment:
I'm curious about this guard, why don't we want to support custom input resources as well?

Reply:
based off your other comment, I had Initially placed this as `HashableInputResource`.